src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/multitable/MultitableFlowsSpec.groovy

```diff
@@ -1510,4 +1447,130 @@ mode with existing flows and hold flows of different table-mode types"() {
+        [involvedSwitches[0], involvedSwitches[2]].each {
+            northbound.updateSwitchProperties(it.dpId, changeSwitchPropsMultiTableValue(initSwProps[it.dpId], true))
+        }
+        [involvedSwitches[1], involvedSwitches[3]].each {
+            northbound.updateSwitchProperties(it.dpId, changeSwitchPropsMultiTableValue(initSwProps[it.dpId], false))
+        }
+        checkDefaultRulesOnSwitches(involvedSwitches)
+        return [involvedSwitches, swPair, desiredPath, initSwProps]
+    }
+
+    def "get triplet of traffgen switches in single table mode with flow path"() {
+        /*
+                def switchPair = topologyHelper.allNotNeighboringSwitchPairs.collectMany { [it, it.reversed] }.find { pair ->
+            desiredPath = pair.paths.find { path ->
+                involvedSwitches = pathHelper.getInvolvedSwitches(path)
+                involvedSwitches.size() == 3 &&
+                        involvedSwitches.every { it.features.contains(SwitchFeature.MULTI_TABLE) }
+            }
+        }
+         */
+        List<PathNode> desiredPath = null
+        List<Switch> involvedSwitches = null
+        def swPair = topologyHelper.allNotNeighboringSwitchPairs
+                .collectMany { [it, it.reversed] }
+                .find{pair ->
+                    desiredPath = pair.paths.find {path ->
+                        involvedSwitches = pathHelper.getInvolvedSwitches(path)
+                        involvedSwitches.size() == 3 &&
+                                involvedSwitches.every {it.features.contains(SwitchFeature.MULTI_TABLE)} &&
+                                [involvedSwitches[0], involvedSwitches[-1]]
+                                        .every {topology.activeTraffGens*.switchConnected.dpId.contains(it.dpId)}
+                    }
+                }
+        assumeTrue(swPair.asBoolean(), 'Unable to find appropriate switch pair')
+        swPair.paths.findAll { it != desiredPath }.each { pathHelper.makePathMorePreferable(desiredPath, it) }
+        Map<SwitchId, SwitchPropertiesDto> initSwProps = involvedSwitches.collectEntries {
+            [(it.dpId): switchHelper.getCachedSwProps(it.dpId)]
+        }
+        withPool {
+            involvedSwitches.eachParallel {
+                northbound.updateSwitchProperties(it.dpId, changeSwitchPropsMultiTableValue(initSwProps[it.dpId], false))
+            }
+        }
+        checkDefaultRulesOnSwitches(involvedSwitches)
+        return [involvedSwitches, swPair, desiredPath, initSwProps]
+
+    }
+    //TODO: move to wrapper class SwitchList.getGarbageRules()
+    def "validate switches"(List<Switch> switchesToValidate) {
+        switchesToValidate.each {sw ->
+            with(northboundV2.validateSwitch(sw.dpId)) { validation ->
+                validation.verifyRuleSectionsAreEmpty(["missing", "excess", "misconfigured"])
+                validation.verifyMeterSectionsAreEmpty(["missing", "excess", "misconfigured"])
```

Comment:
Note: in V2 validation we have field `asExpected`. If it's true - everything is ok on a switch. you can you this field

Reply:
Updated this method. Didn't go through all the tests though.